browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -162,6 +163,21 @@ bool ApplyPotentialReferrerBlock(std::shared_ptr<BraveRequestInfo> ctx) {
   return false;
 }
 
+bool ShouldBlockOnionAddress(std::shared_ptr<BraveRequestInfo> ctx) {
+  if (ctx->browser_context && ctx->browser_context->IsTor()) {
+    // Allow .onion addresses in Tor windows.
+    return false;
+  }
+  if (!ctx->request_url.has_host()) {
+    return false;
+  }
+  // TODO: don't block "Open in Tor" button (unless #if BUILDFLAG(ENABLE_TOR) is false)
```

Comment:
directly adding `.onion` address doesn't have "Open in Tor" button currently(only `onion-location` header), we can add that button for this case though.